src/main/scala/net/cakesolutions/CakeDockerUtils.scala

```diff
@@ -0,0 +1,161 @@
+
+  }
+
+  /**
+    * Dumping logs for all containers defined in docker-compose scope
+    * @param ymlFiles docker-compose files which defines scope
+    * @param targetDir target directory for log files
+    * @param log sbt logger
+    */
+  def dumpLogs(ymlFiles: Seq[File], targetDir: File)(
+    implicit log: Logger
+  ): Unit = {
+    containerNames(ymlFiles).foreach { container =>
+      val dockerTargetDir = file(targetDir.getAbsolutePath + "/docker")
+      log.info(
+        s"Dumping container: $container logs " +
+          s"to ${dockerTargetDir.getAbsolutePath}."
+      )
+      IO.createDirectory(dockerTargetDir)
+      val logFile = file(dockerTargetDir.getAbsolutePath + s"/$container.log")
+
+      Try(Seq("docker", "logs", container) ! new FileLogger(logFile)) match {
+        case Success(_) =>
+          log.success(s"Dumped logs of $container successfully.")
+        case Failure(e) =>
+          log.error(s"Failed to fetch logs for $container: $e")
+      }
+    }
+  }
+
+  private class FileLogger(logFile: File) extends ProcessLogger {
+    IO.write(logFile, "")
+
+    def buffer[T](f: => T): T = f
+
+    def error(s: => String): Unit =
+      IO.append(logFile, s"$s\n".getBytes("utf-8"))
+
+    def info(s: => String): Unit =
+      IO.append(logFile, s"$s\n".getBytes("utf-8"))
+  }
+
+  private def containerNames(
+    ymlFiles: Seq[File]
+  )(implicit log: Logger): Seq[String] = {
+
+    val projectOverrides =
+      ymlFiles.flatMap(yaml => Seq("-f", yaml.getCanonicalPath))
+
+    val listContainers =
+      Seq("docker-compose") ++ projectOverrides ++ Seq("ps", "-q")
+
+    def containerName(id: String) =
+      Seq("docker", "ps", "-a", "--format", "{{.Names}}", "-f", s"id=$id")
+
+    Try(listContainers.!!.trim.split("\\s").toSeq) match {
+      case Failure(e) =>
+        log.error(s"Failed to fetch container identities: $e")
+        Seq.empty[String]
+      case Success(ids) =>
+        ids.filter(_.nonEmpty).flatMap { id =>
+          Try(containerName(id).!!.trim) match {
+            case Success(containerName) =>
+              Some(containerName)
+            case Failure(_) =>
+              log.error(s"Failed to fetch container name: $id")
+              None
+          }
+        }
+    }
+  }
+
+  /**
+    * A helper ADT to represent docker version information.
+    */
+  sealed case class Version(major: Int, minor: Int) {
+
+    def gte(that: Version): Boolean = {
+      major > that.major || (major == that.major && minor >= that.minor)
+    }
+
+    def checkDockerCompose(minVersion: Version): Boolean = gte(minVersion)
+
+    def checkDocker(minVersion: Version): Boolean = gte(minVersion)
+
+  }
+
+  object Version {
+
+    private val versionRegex = """\s([0-9]+)\.([0-9]+)[ .,-]""".r
+
+    def parse(text: String): Version = {
+      versionRegex findFirstIn text match {
+        case Some(versionRegex(majorStr, minorStr)) =>
+          Version(majorStr.toInt, minorStr.toInt)
+        case None =>
```

Comment:
We should avoid mapping errors to default values - this can lead to potential issues in the future if folk wish to reuse this code. It is far better (and more flexible) to map directly to an explicit error value here.